packages/ethereum-contracts/contracts/agreements/AgreementLibrary.sol

```diff
@@ -140,17 +140,35 @@ library AgreementLibrary {
 
                 appContext = ISuperfluid(msg.sender).decodeCtx(newCtx);
 
-                // adjust allowance used to the range [appAllowanceWanted..appAllowanceGranted]
-                appContext.appAllowanceUsed = max(0, min(
-                    inputs.appAllowanceGranted.toInt256(),
-                    max(appContext.appAllowanceWanted.toInt256(), appContext.appAllowanceUsed)));
-
+                // adjust credit used to the range [appCreditUsed..appCreditGranted]
+                appContext.appCreditUsed = _adjustNewAppCreditUsed(
+                    inputs.appCreditGranted,
+                    appContext.appCreditUsed
+                );
             }
             // [SECURITY] NOTE: ctx should be const, do not modify it ever to ensure callback stack correctness
-            newCtx = _popCallbackStack(ctx, appContext.appAllowanceUsed);
+            newCtx = _popCallbackStack(ctx, appContext.appCreditUsed);
         }
     }
 
+    /**
+     * Determines how much app credit the app will use.
+     * @param appCreditGranted set prior to callback based on input flow
+     * @param appCreditUsed set in callback - summed deposit deltas of agreements
+     */
+    function _adjustNewAppCreditUsed(
+        uint256 appCreditGranted,
+        int256 appCreditUsed
+    ) internal pure returns (int256) {
+        return max(
+            0,
+            min(
+                appCreditGranted.toInt256(),
+                appCreditUsed
```

Comment:
1. add comment: appCreditUsed can be minus, hence `max(0, -)` needed
2. possible other names for appCreditUsed: app[CallbackSession]DepositDelta, comment in ISuperfluid interface about what does it mean. Make some more elaboration of its use per discussion.

Reply:
⚠️ new logic